restapi/api/device_endpoints.go

```diff
@@ -115,5 +117,147 @@ func ResetLocation(c *gin.Context) {
 		c.JSON(http.StatusInternalServerError, GenericResponse{Error: err.Error()})
 		return
 	}
+
 	c.JSON(http.StatusOK, GenericResponse{Message: "Device location reset"})
 }
+
+//========================================
+// DEVICE STATE CONDITIONS
+//========================================
+
+var deviceConditionsMap = make(map[string]deviceCondition)
```

Comment:
I have added `LimitNumClientsUDID()` as a Middleware so you can limit the max. calls to a device specific endpoint to what makes sense. My idea was for every endpoint that should never be executed in parallel, to use that middleware.  Right now it is even hardcoded to 1 max. concurrent call. If you put that in front, you do not need a mutex and can simplify your code.

Reply:
In general for most functionality, allowing to call it in parallel is probably a bad idea as the behavior of the device would be undefined. Maybe the middleware must be even extended at some point to prevent calling f.ex. disable and enable at the same time.

Reply:
The purpose of this map is to hold the device conditions for all host devices that have them applied. In this case the Mutex is introduced to handle the concurrent access to the map for different devices enabling/disabling of conditions. I can still use that middleware but it will not simplify the current code :)

Reply:
So I think we need a middleware to limit concurrent access based on udid level. I cannot verify it, but I think just attaching the `LimitNumClientsUDID` middleware to a routergroup should already lock it down, so any endpoint in that group can only be called in series. 
Then you don't need a Mutex because there cannot be any concurrent calls. 
Doing this with such a Middleware is preferrable imo because it is less error prone than adding Mutexes to all endpoints.